### PR TITLE
feat: implement filesize validation for movies and episodes

### PR DESCRIPTION
**Pull request by davidemarcoli**
Don't remove it again @Gaisberg! 😡

# Pull Request Check List

Resolves: #issue-number-here

- [ ] Added **tests** for changed code.
- [ ] Updated **documentation** for changed code.

## Summary by CodeRabbit

- **New Features**
	- Enhanced file validation for movies and shows based on acceptable file sizes.
	- New functions introduced to check file size validity for movies and shows.
	- Introduced enumerations for media types to improve type safety and clarity.

- **Bug Fixes**
	- Improved error handling for rate limits in request processing.

- **Refactor**
	- Simplified file size validation logic by transitioning from instance variables to global functions and variables.
	- Updated minimum file size constraints to enforce non-negative limits for movies and episodes.
	- Ensured consistency in service results by converting keys to lowercase in the scraping process.

### CODE REVIEW

File: src/program/services/downloaders/shared.py [hunk 1]
```diff
@@ -162,3 +117,48 @@ def hash_from_uri(magnet_uri: str) -> str:
...
+        return False
+    if not isinstance(settings_manager.settings.downloaders.episode_filesize_mb_min, int) or settings_manager.settings.downloaders.episode_filesize_mb_min < -1:
+        logger.error("Episode filesize min is not set or invalid.")
+        return False
+    if not isinstance(settings_manager.settings.downloaders.episode_filesize_mb_max, int) or settings_manager.settings.downloaders.episode_filesize_mb_max < -1:
+        logger.error("Episode filesize max is not set or invalid.")
+        return False
+    return True
+
+are_filesizes_valid = _validate_filesizes()
```
[the remark this answers is not in the document]
Comment (dreulavelle) [reply src/program/services/downloaders/shared.py]: this looks good! @davidemarcoli

File: src/program/services/downloaders/__init__.py [hunk 2]
```diff
@@ -90,21 +91,24 @@ def update_item_attributes(self, item: MediaItem, info, container) -> bool:
...
+            if item.type == "movie" and self.service.file_finder.container_file_matches_movie(file) and filesize_is_acceptable_movie(file[self.service.file_finder.filesize_attr]):
                 item.file = file[self.service.file_finder.filename_attr]
                 item.folder = info["filename"]
                 item.alternative_folder = info["original_filename"]
                 item.active_stream = {"infohash": info["hash"], "id": info["id"]}
                 found = True
                 break
+            else:
+                logger.debug(f"Deleting {info['id']} because it doesn't match the item type or size")
+                self.delete_torrent(info["id"])
```
Comment (Gaisberg) [inline src/program/services/downloaders/__init__.py:103]: Please handle this outside the method

File: src/program/services/downloaders/__init__.py [hunk 3]
```diff
@@ -116,4 +120,7 @@ def update_item_attributes(self, item: MediaItem, info, container) -> bool:
                             # We have to make sure the episode is correct if item is an episode
                             if item.type != "episode" or (item.type == "episode" and episode.number == item.number):
                                 found = True
+                else:
+                    logger.debug(f"Deleting {info['id']} because it doesn't match the item type or size")
+                    self.delete_torrent(info["id"])
```
Comment (Gaisberg) [inline src/program/services/downloaders/__init__.py:125]: same as above

File: src/program/services/downloaders/shared.py [hunk 4]
```diff
@@ -162,3 +118,75 @@
...
+        return True
+        
+    min_mb, max_mb = _get_size_limits(media_type)
+    min_size = 0 if min_mb == -1 else _convert_to_bytes(min_mb)
+    max_size = float("inf") if max_mb == -1 else _convert_to_bytes(max_mb)
+    
+    is_acceptable = min_size <= filesize <= max_size
+    if not is_acceptable:
+        logger.debug(f"{media_type} filesize {filesize} bytes is not within acceptable range {min_size} - {max_size} bytes")
+    return is_acceptable
```
[the remark this answers is not in the document]
Comment (Gaisberg) [reply src/program/services/downloaders/shared.py:185]: Please take a look at coderabbitai's response

**Review by Gaisberg — CHANGES_REQUESTED** [review]

File: src/program/media/item.py [hunk 5]
```diff
@@ -17,6 +18,22 @@
...
+class MovieMediaType(Enum):
+    """Media types"""
+    Movie = "movie"
+
+class MediaType(Enum):
+    """Combined media types"""
+    Show = ShowMediaType.Show.value
+    Season = ShowMediaType.Season.value
+    Episode = ShowMediaType.Episode.value
+    Movie = MovieMediaType.Movie.value
```
Comment (dreulavelle) [inline src/program/media/item.py:36]: we dont need these classes lol, no need to make it more complicated than it has to be! ;)

**Review by Gaisberg — APPROVED** [review]
LGTM! Props for the type classes

**Pull request merged**